src/pudl/analysis/plant_parts_eia.py

```diff
@@ -0,0 +1,1733 @@
+    'plant_technology',
+    'plant_prime_fuel',
+    'plant_ferc_acct',
+    'plant_gen'
+]
+
+
+IDX_TO_ADD: List[str] = ['report_date', 'operational_status_pudl']
+"""
+list: list of additional columns to add to the id_cols in :py:const:`PLANT_PARTS`.
+The id_cols are the base columns that we need to aggregate on, but we also need
+to add the report date to keep the records time sensitive and the
+operational_status_pudl to separate the operating plant-parts from the
+non-operating plant-parts.
+"""
+
+IDX_OWN_TO_ADD: List[str] = ['utility_id_eia', 'ownership']
+"""
+list: list of additional columns beyond the :py:const:`IDX_TO_ADD` to add to the
+id_cols in :py:const:`PLANT_PARTS` when we are dealing with plant-part records
+that have been broken out into "owned" and "total" records for each of their
+owners.
+"""
+
+SUM_COLS: List[str] = [
+    'total_fuel_cost',
+    'net_generation_mwh',
+    'capacity_mw',
+    'capacity_eoy_mw',
+    'total_mmbtu',
+]
+"""
+list: list of columns to sum when aggregating a table.
+"""
+
+WTAVG_DICT = {
+    'fuel_cost_per_mwh': 'capacity_mw',
+    'heat_rate_mmbtu_mwh': 'capacity_mw',
+    'fuel_cost_per_mmbtu': 'capacity_mw',
+}
+"""
+dict: a dictionary of columns (keys) to perform weighted averages on and
+the weight column (values)"""
+
+
+CONSISTENT_ATTRIBUTE_COLS = [
+    'fuel_type_code_pudl',
+    'planned_retirement_date',
+    'retirement_date',
+    'generator_id',
+    'unit_id_pudl',
+    'technology_description',
+    'energy_source_code_1',
+    'prime_mover_code',
+    'ferc_acct_name',
+]
+"""
+list: a list of column names to add as attributes when they are consistent into
+the aggregated plant-part records.
+"""
+
+PRIORITY_ATTRIBUTES_DICT = {
+    'operational_status': ['existing', 'proposed', 'retired'],
+}
+
+
+FIRST_COLS = ['plant_id_eia', 'report_date', 'plant_part', 'generator_id',
+              'unit_id_pudl', 'prime_mover_code', 'energy_source_code_1',
+              'technology_description', 'ferc_acct_name',
+              'utility_id_eia', 'true_gran', 'appro_part_label']
+
+
+class MakeMegaGenTbl(object):
+    """
+    Compiler for a MEGA generator table with ownership integrated.
+
+    Examples
+    --------
+    **Input Tables**
+
+    Here is an example of one plant with three generators. We will use
+    ``capacity_mw`` as the data column.
+
+    >>> df_mcoe = pd.DataFrame({
+    ...     'plant_id_eia': [1, 1, 1],
+    ...     'generator_id': ['a', 'b', 'c'],
+    ...     'capacity_mw': [100, 100, 200]
+    ... })
+
+    The ownership table from EIA 860 includes one record for every owner of
+    each generator. In this example generator ``c`` has two owners.
+
+    >>> df_own_eia860 = pd.DataFrame({
+    ...     'plant_id_eia': [1, 1, 1, 1],
+    ...     'generator_id': ['a', 'b', 'c', 'c'],
+    ...     'utility_id_eia': [111, 111, 111, 111],
+    ...     'owner_utility_id_eia': [111, 111, 111, 888],
+    ...     'fraction_owned': [1, 1, .75, .25]
+    ... })
```

Comment:
You could include the method calls so folks understand what is processing the inputs and outputs here.  I don't know if this is accurate but something like this: 

```python
gens_mega_no_owners = self.get_gens_mega_table(df_mcoe)
gens_mega = self.slice_by_ownership(gens_mega_no_owners, df_own_eia860)
```